src/createServer.js

```diff
@@ -1,8 +1,51 @@
 'use strict';
 
+const http = require('http');
+const fs = require('fs');
+const url = require('url');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = http.createServer((req, res) => {
+    const { pathname } = new url.URL(req.url, `http://${req.headers.host}`);
```

Comment:
Using `url.URL` is correct, but note that `url.parse` is deprecated. Ensure that `req.url` is a valid URL string, especially if it might contain special characters. Consider using `decodeURIComponent` on `req.url` if necessary.